src/distributions/float.rs

```diff
@@ -144,63 +146,173 @@ float_impls! { f32, u32, 23, 127 }
 float_impls! { f64, u64, 52, 1023 }
 
 
+#[cfg(feature="simd_support")]
+macro_rules! simd_float_impls {
+    ($ty:ident, $uty:ident, $f_scalar:ty, $u_scalar:ty,
+     $fraction_bits:expr, $exponent_bias:expr) => {
+        impl IntoFloat for $uty {
+            type F = $ty;
+            #[inline(always)]
+            fn into_float_with_exponent(self, exponent: i32) -> $ty {
+                // The exponent is encoded using an offset-binary representation
+                let exponent_bits: $u_scalar =
+                    (($exponent_bias + exponent) as $u_scalar) << $fraction_bits;
+                unsafe { mem::transmute(self | $uty::splat(exponent_bits)) }
+            }
+        }
+
+        impl Distribution<$ty> for Standard {
+            fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> $ty {
+                // Multiply-based method; 24/53 random bits; [0, 1) interval.
+                // We use the most significant bits because for simple RNGs
+                // those are usually more random.
+                let float_size = mem::size_of::<$f_scalar>() * 8;
+                let precision = $fraction_bits + 1;
+                let scale = $ty::splat(1.0 / ((1 as $u_scalar << precision) as $f_scalar));
```

Comment:
Note: most arithmetic and bitwise operations allow you to just use scalar values: 
```rust
let scale = $ty::splat(1.0 / ((1 as $u_scalar << precision) as $f_scalar));
scale * value
// is the same as
let scale = 1.0 / ((1 as $u_scalar << precision) as $f_scalar);
scale * value